test/e2e/chaos/basic.go

```diff
@@ -1640,151 +1660,262 @@ selector:
 
-				err = cli.Delete(ctx, networkDelay.DeepCopy())
+				result = probeNetworkCondition(c, networkPeers, ports)
+				framework.ExpectEqual(len(result[networkConditionBlocked]), 0)
+				framework.ExpectEqual(result[networkConditionSlow], [][]int{{0, 1}, {0, 2}})
+
+				err = cli.Delete(ctx, networkDelayWithTarget.DeepCopy())
 				framework.ExpectNoError(err, "delete network chaos error")
 				time.Sleep(5 * time.Second)
-				framework.ExpectEqual(allSlowConnection(), [][]int{{0, 2}})
+
+				result = probeNetworkCondition(c, networkPeers, ports)
+				framework.ExpectEqual(len(result[networkConditionBlocked]), 0)
+				framework.ExpectEqual(result[networkConditionSlow], [][]int{{0, 2}})
+
 				err = cli.Delete(ctx, evenNetworkDelay.DeepCopy())
 				framework.ExpectNoError(err, "delete network chaos error")
 				time.Sleep(5 * time.Second)
-				framework.ExpectEqual(len(allSlowConnection()), 0)
 
-				complicateNetem := &v1alpha1.NetworkChaos{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      "network-chaos-3",
-						Namespace: ns,
-					},
-					Spec: v1alpha1.NetworkChaosSpec{
-						Action: v1alpha1.DelayAction,
-						Selector: v1alpha1.SelectorSpec{
-							Namespaces:     []string{ns},
-							LabelSelectors: map[string]string{"app": "network-peer-0"},
-						},
-						Mode: v1alpha1.OnePodMode,
-						TcParameter: v1alpha1.TcParameter{
-							Delay: &v1alpha1.DelaySpec{
-								Latency:     "200ms",
-								Correlation: "25",
-								Jitter:      "0ms",
-							},
-							Loss: &v1alpha1.LossSpec{
-								Loss:        "25",
-								Correlation: "25",
-							},
-							Duplicate: &v1alpha1.DuplicateSpec{
-								Duplicate:   "25",
-								Correlation: "25",
-							},
-							Corrupt: &v1alpha1.CorruptSpec{
-								Corrupt:     "25",
-								Correlation: "25",
-							},
-						},
-						Duration: pointer.StringPtr("9m"),
-						Scheduler: &v1alpha1.SchedulerSpec{
-							Cron: "@every 10m",
-						},
-					},
-				}
-				klog.Infof("Injecting delay for 0")
+				result = probeNetworkCondition(c, networkPeers, ports)
+				framework.ExpectEqual(len(result[networkConditionBlocked]), 0)
+				framework.ExpectEqual(len(result[networkConditionSlow]), 0)
+
+				complicateNetem := makeNetworkDelayChaos(
+					ns, "network-chaos-3",
+					map[string]string{"app": "network-peer-0"},
+					nil, // no target specified
+					v1alpha1.OnePodMode,
+					v1alpha1.OnePodMode,
+					testDelayTcParamEvenMoreComplicate,
+					testDelayDuration,
+					testDelaySchedulerSpec,
+				)
+
+				klog.Infof("Injecting delay for 0 with more complicate netem")
 				err = cli.Create(ctx, complicateNetem.DeepCopy())
 				framework.ExpectNoError(err, "create network chaos error")
 				time.Sleep(5 * time.Second)
-				framework.ExpectEqual(allSlowConnection(), [][]int{{0, 1}, {0, 2}, {0, 3}})
+
+				result = probeNetworkCondition(c, networkPeers, ports)
+
+				framework.ExpectEqual(len(result[networkConditionBlocked]), 0)
```

Comment:
@STRRL After merged into `probeNetworkCondition`, I found that the `couldConnect` network test  would broken, since UDP is an unreliable protocol under the bad network.

From now, I workaround this, print out the log and still return true, so it's still 0 at here.

Reply:
You mean `time.Sleep(2 * time.Second)` between `sendUDPPacket` and `recvUDPPacket`?

Reply:
Nope. In the previous, we didn't do `couldConnect` test at here, we called `allSlowConnection` and it ONLY did `testDelay`, as we all known, `testDelay` would send HTTP post request, which is a TCP-based request, and `couldConnect` is an UDP request.

At here, we're testing the complicate netem delay case, and it would make each network request may lost packet, in other words, it may also make the `couldConnect` return bad result. After we make the two functions merged into `probeNetworkCondition`, I met this issue! `couldConnect` may receive empty data(lost since it's UDP request), and I workaround it, see [this line](https://github.com/chaos-mesh/chaos-mesh/pull/1069/files#diff-6370badef8e5715e4f72fad6d6488b517cd1cca6f98764aa93209e5d3710e36bR2188).

Reply:
I guess that this issue ONLY happens on `testDelayTcParamEvenMoreComplicate`, does it?  
It makes delay, loss, duplicate and corrupt at same time. Honestly I have no idea for making assertion stable under this situation. 🙈

Reply:
> I guess that this issue ONLY happens on testDelayTcParamEvenMoreComplicate, does it?
Yes, so I choose workaround :(